extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -112,14 +112,25 @@ export class JupyterServerInstallation implements IJupyterServerInstallation {
 	private readonly _requiredKernelPackages: Map<string, PythonPkgDetails[]>;
 	private readonly _requiredPackagesSet: Set<string>;
 
-	constructor(extensionPath: string, outputChannel: OutputChannel, apiWrapper: ApiWrapper, pythonInstallationPath?: string) {
+	public readonly runningOnSAW: boolean;
+
+	constructor(extensionPath: string, outputChannel: vscode.OutputChannel, apiWrapper: ApiWrapper) {
 		this.extensionPath = extensionPath;
 		this.outputChannel = outputChannel;
 		this.apiWrapper = apiWrapper;
-		this._pythonInstallationPath = pythonInstallationPath || JupyterServerInstallation.getPythonInstallPath(this.apiWrapper);
+
+		this.runningOnSAW = vscode.env.appName.toLowerCase().indexOf('saw') > 0;
```

Comment:
I think it'd be better to just have the quality be a context set by the core at startup - that's something that would be beneficial to have generally available anyways. 

Look at how the platform keys are set for how this would be done

https://github.com/microsoft/azuredatastudio/blob/main/src/vs/platform/contextkey/common/contextkeys.ts#L10